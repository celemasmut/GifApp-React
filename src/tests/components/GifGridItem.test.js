import { shallow } from 'enzyme'
import React from 'react';
import { GifGridItem } from '../../components/GifGridItem';


describe('Prueba de GifGridItem', () => {
    
    const title = 'mi titulo';
    const url = 'https://localhost/algo.jpg';
    const wrapper = shallow(<GifGridItem title= {title} url= {url}/>)
    
    test('debe mostrar el componente <GifGridItem />> correctamente', () => {
        
        expect( wrapper ).toMatchSnapshot();


    });

    test('debe tener un parrafo con el title', () => {

        const parrafo = wrapper.find('p');
        expect(parrafo.text().trim() ).toBe(title);
        
    });

    test('debe de tener una imagen igual al url y al alt de los props', () => {

        const img = wrapper.find('img');

        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
        
    });

    test('debe tener animate__headShake', () => {
        const div = wrapper.find('div');

        const className = div.prop('className');

        expect(className.includes('animate__headShake')).toBe(true);
    })
    
    
    
    
})
