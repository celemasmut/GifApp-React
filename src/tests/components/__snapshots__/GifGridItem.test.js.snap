// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba de GifGridItem debe mostrar el componente <GifGridItem />> correctamente 1`] = `
<div
  className="card animate__animated animate__headShake"
>
  <img
    alt="mi titulo"
    src="https://localhost/algo.jpg"
  />
  <p>
    mi titulo
  </p>
</div>
`;
